### PR TITLE
feat(driver-adapter-wasm): add d1-vitest suite

### PR DESCRIPTION
**Pull request by jkomyno**
This PR complements https://github.com/prisma/prisma/pull/24554.

### CODE REVIEW

File: driver-adapters-wasm/d1-vitest/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "name": "Alberto Schiabel",
+    "email": "schiabel@prisma.io"
+  },
+  "scripts": {
+    "test": "NODE_OPTIONS='--experimental-vm-modules' vitest --run"
+  },
+  "license": "MIT",
+  "dependencies": {
+    "@prisma/adapter-d1": "5.16.0-integration-vitest-workaround.1",
+    "@prisma/client": "5.16.0-integration-vitest-workaround.1"
```
Comment (jkomyno) [inline driver-adapters-wasm/d1-vitest/package.json]: TODO: downgrade to `5.16.0-dev.60` before merging.

File: driver-adapters-wasm/d1-vitest/run.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -eu
+
+pnpm install
+pnpm prisma generate
+
+# Vitest is run locally, we don't need to deploy the database migration
+pnpm wrangler d1 migrations apply d1-vitest --local
```
Comment (Druue) [inline driver-adapters-wasm/d1-vitest/run.sh:9]: `d1 migrations apply` already defaults to local fwiw
https://developers.cloudflare.com/workers/wrangler/commands/#migrations-apply

File: driver-adapters-wasm/d1-vitest/src/user.test.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { env } from "cloudflare:test"
+import { getPrisma } from "./prismaClient"
+
+describe('user model', () => { 
+  let prisma = getPrisma(env.D1_DATABASE)
+
+  beforeEach(async () => {
+    await prisma.user.deleteMany()
```
Comment (Druue) [inline driver-adapters-wasm/d1-vitest/src/user.test.ts:8]: Check failures on this for some reason with:
```
{ extended_code: 1, message: Some("D1_ERROR: no such table: main.User") }
```

[Example run](https://github.com/prisma/ecosystem-tests/actions/runs/9646280592/job/26602506669?pr=5140#step:8:641)

File: driver-adapters-wasm/d1-vitest/run.sh (hunk 2 above)
Comment (jkomyno) [reply driver-adapters-wasm/d1-vitest/run.sh:9]: Yes, but I'd rather have explicit commands than using implicit defaults, for the sake of future clarity and to remark the local-first intent. Would you agree?

**Comment by jkomyno** [comment]
The newly added `driver-adapters-wasm` test, `d1-vitest`, now works: https://github.com/prisma/ecosystem-tests/pull/5140#discussion_r1651005735.

Following https://github.com/prisma/ecosystem-tests/pull/5140#discussion_r1651005735, I'm bumping the Prisma Client versions to align them to the latest `dev` version. This will cause `d1-vitest` to temporarily fail until it's merged into the `dev` branch.

File: driver-adapters-wasm/d1-vitest/run.sh (hunk 2 above)
Comment (Druue) [reply driver-adapters-wasm/d1-vitest/run.sh:9]: Sounds good to me :)

**Review by Druue — APPROVED** [review]

**Pull request merged**